modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/TranslationManager.java

```diff
@@ -38,30 +39,30 @@
 /**
  * @author Yasuyuki Takeo
  */
-@Component(immediate = true, service = DeepLClient.class)
-public class DeepLClient {
+@Component(immediate = true, service = TranslationManager.class)
+public class TranslationManager {
 
-	public TranslateResponse execute(
+	public List<Translation> getTranslateResponse(
 			String authKey, String text, String sourceLanguageId,
 			String targetLanguageId, String url)
 		throws IOException, PortalException {
 
-		return JSONUtil.toObject(
-			_fetch(authKey, text, sourceLanguageId, targetLanguageId, url),
-			TranslateResponse.class);
-	}
-
-	public List<SupportedLanguage> getSupportedLanguages(
-			String authKey, String target, String url)
-		throws IOException, PortalException {
-
-		return JSONUtil.toObject(
-			_getSupportedLanguage(authKey, target, url),
-			new TypeReference<List<SupportedLanguage>>() {
-			});
+		JSONObject jsonObject = _jsonFactory.createJSONObject(
+			post(authKey, text, sourceLanguageId, targetLanguageId, url));
+
+		return JSONUtil.toList(
+			jsonObject.getJSONArray("translations"),
+			customFieldJSONObject -> new Translation() {
+				{
+					detectedSourceLanguage = customFieldJSONObject.getString(
+						"detected_source_language");
+					text = customFieldJSONObject.getString("text");
+				}
+			},
+			_log);
```

Comment:
You should not use dynamic initializers unless strictly necessary. Please add a constructor to Translation that takes the detectedSourceLanguage and the text.

Reply:
@adolfopa but the code BChan pointed out using [this way though](https://github.com/liferay/liferay-portal/blob/8b6306ce5ca9aa8ca896ba2223864897c5488b55/modules/apps/digital-signature/digital-signature-impl/src/main/java/com/liferay/digital/signature/internal/manager/DSCustomFieldManagerImpl.java#L42-L61)), how come?

Reply:
That code is not using a dynamic init. block. You need to either use setters or a constructor.

Reply:
@adolfopa Oh, yeah, it was the interface, got it.